crates/bevy_asset/src/processor/mod.rs

```diff
@@ -514,34 +512,36 @@ impl AssetProcessor {
 
         /// Retrieves asset paths recursively. If `clean_empty_folders_writer` is Some, it will be used to clean up empty
         /// folders when they are discovered.
-        fn get_asset_paths<'a>(
+        async fn get_asset_paths<'a>(
             reader: &'a dyn ErasedAssetReader,
             clean_empty_folders_writer: Option<&'a dyn ErasedAssetWriter>,
             path: PathBuf,
             paths: &'a mut Vec<PathBuf>,
-        ) -> BoxedFuture<'a, Result<bool, AssetReaderError>> {
-            Box::pin(async move {
-                if reader.is_directory(&path).await? {
-                    let mut path_stream = reader.read_directory(&path).await?;
-                    let mut contains_files = false;
-                    while let Some(child_path) = path_stream.next().await {
-                        contains_files =
-                            get_asset_paths(reader, clean_empty_folders_writer, child_path, paths)
-                                .await?
-                                && contains_files;
-                    }
-                    if !contains_files && path.parent().is_some() {
-                        if let Some(writer) = clean_empty_folders_writer {
-                            // it is ok for this to fail as it is just a cleanup job.
-                            let _ = writer.remove_empty_directory(&path).await;
-                        }
+        ) -> Result<bool, AssetReaderError> {
+            if reader.is_directory(&path).await? {
+                let mut path_stream = reader.read_directory(&path).await?;
+                let mut contains_files = false;
+                while let Some(child_path) = path_stream.next().await {
+                    contains_files = Box::pin(get_asset_paths(
+                        reader,
+                        clean_empty_folders_writer,
+                        child_path,
+                        paths,
+                    ))
+                    .await?
+                        && contains_files;
```

Comment:
Based on [the description of `&&`](https://doc.rust-lang.org/reference/expressions/operator-expr.html#lazy-boolean-operators), it actually serves no purpose beyond a normal `&` in this instance, therefore the `&=` operator can be used instead, simplifying the expression without any change. This will also avoid the unintended consequence of not evaluating `get_asset_paths` if the order of the `&&` operands were reversed.

More importantly, I think it's actually _wrong_. I believe the intent was to use logical OR, since (I believe) this `get_asset_paths` function is supposed to return `true` if _any_ sub-folder contains a file, but right now it will only do so if _every_ folder does, which doesn't seem right to me.

I believe this is why line 536 has a comment suppressing the error likely being returned by `remove_empty_directory`.

So, with _all_ that context, here's my suggestion:

```suggestion
                    contains_files |= Box::pin(get_asset_paths(
                        reader,
                        clean_empty_folders_writer,
                        child_path,
                        paths,
                    ))
                    .await?;
```

Reply:
Ohh nice catch! Yeah I think it's definitely meant to be an or. 

I've made this change here now, but let me know if that's better as a seperate PR.

Reply:
Just update the title and description to capture the change please :)

Reply:
Sweet! Update both, hopefully this works.